### PR TITLE
feat: working consumption api with plan core

### PR DESCRIPTION
**Pull request by callicles**
Adds the consumption api to the core

### CODE REVIEW

**Review by tg339 — APPROVED** [review]

File: apps/framework-cli/src/cli/display.rs [hunk 1]
```diff
@@ -340,6 +340,18 @@ pub fn show_changes(infra_plan: &InfraPlan) {
...
             }
+            ProcessChange::ConsumptionApiWebServer(Change::Added(_)) => {
+                infra_added("Starting Consumption WebServer...");
+            }
+            ProcessChange::ConsumptionApiWebServer(Change::Removed(_)) => {
+                infra_removed("Stoping Consumption WebServer...");
+            }
+            ProcessChange::ConsumptionApiWebServer(Change::Updated {
+                before: _,
+                after: _,
```
Comment (phiSgr) [inline apps/framework-cli/src/cli/display.rs]: Just write `Change::Updated { .. }`?

File: apps/framework-cli/src/cli/watcher.rs [hunk 2]
```diff
@@ -340,28 +339,41 @@ async fn watch(
...
+                                    Ok(())
+                                }
+                                Err(e) => {
+                                    show_message!(MessageType::Error, {
+                                        Message {
+                                            action: "\nFailed".to_string(),
+                                            details: format!("\n {}", e),
+                                        }
+                                    });
+                                    Ok(())
```
Comment (phiSgr) [inline apps/framework-cli/src/cli/watcher.rs]: Both branches return `Ok(())`
what about moving that outside the pattern matching?

File: apps/framework-cli/src/cli/display.rs (hunk 1 above)
Comment (callicles) [reply apps/framework-cli/src/cli/display.rs]: Thanks did not know this one!

File: apps/framework-cli/src/cli/watcher.rs (hunk 2 above)
Comment (callicles) [reply apps/framework-cli/src/cli/watcher.rs]: yep - good catch

**Pull request merged**